File: comparison/results/haiku/baseline/humaneval/HumanEval_1/python/solution.py
from typing import List


def separate_paren_groups(paren_string: str) -> List[str]:
    """ Input to this function is a string containing multiple groups of nested parentheses. Your goal is to
    separate those group into separate strings and return the list of those.
    Separate groups are balanced (each open brace is properly closed) and not nested within each other
    Ignore any spaces in the input string.
    >>> separate_paren_groups('( ) (( )) (( )( ))')
    ['()', '(())', '(()())']
    """
    s = paren_string.replace(' ', '')
    
    result = []
    balance = 0
    current = ''
    
    for char in s:
        if char == '(':
            balance += 1
        elif char == ')':
            balance -= 1
        
        current += char
        
        if balance == 0:
            result.append(current)
            current = ''
    
    return result

File: comparison/results/haiku/baseline/humaneval/HumanEval_1/go/solution.go
func separate_paren_groups(paren_string string) []string {
	var result []string
	var current string
	depth := 0
	
	for _, ch := range paren_string {
		if ch == ' ' {
			continue
		}
		
		current += string(ch)
		if ch == '(' {
			depth++
		} else if ch == ')' {
			depth--
		}
		
		if depth == 0 && len(current) > 0 {
			result = append(result, current)
			current = ""
		}
	}
	
	return result
}
